e2e/testcases/custom_resources_test.go

```diff
@@ -109,16 +109,13 @@ func TestCRDDeleteBeforeRemoveCustomResourceV1(t *testing.T) {
 		nt.T.Fatal(err)
 	}
 
-	// Wait for remediator to detect the drift (managed Anvil CR was deleted)
-	// and record it as a conflict.
-	// TODO: distinguish between management conflict (spec/generation drift) and concurrent status update conflict (resource version change)
-	err = nomostest.ValidateMetrics(nt,
+	// Reverting a manual change is NOT considered a "conflict".
+	// Conflict implies either a different reconciler manager or drift between GET (or watch event) & UPDATE.
+	// TODO: distinguish between management conflict (unexpected manager annotation) and resource conflict (resource version change)
+	nt.Must(nomostest.ValidateMetrics(nt,
 		nomostest.ReconcilerErrorMetrics(nt, rootSyncLabels, firstCommitHash, metrics.ErrorSummary{
-			Conflicts: 1,
-		}))
-	if err != nil {
-		nt.T.Fatal(err)
-	}
+			Conflicts: 0,
```

Comment:
Why can this pass with `Conflict: 0` or `Conflict: 1` with no code changes? Seems a bit suspicious

Reply:
Looking through the code again, I think this is actually a race condition. But I was also confused because the conflict metric is overloaded. It's being used to mean 4 different things:
1. Remediator got StatusReasonNotFound when performing an Update. This is a race condition, with another controller deleting an object before the Remediator had a chance to perform an update event in response to a previous update event.
2. Remediator got StatusReasonConflict when performing an Update. This just means some object of the same resource type got updated between event received and event handled. It may or may not be the same object at all.
3. Remediator observed a management conflict (manager annotation says some other reconciler is managing the object). This only gets reported by root reconcilers today.

This is problematic because:
- It confuses management conflicts with resource conflicts
- It doesn't record the metric reason #3 at all from namespace reconcilers
- It doesn't record the metric from the applier, even though it can report the same management conflict error

The problem with this particular test is with reason #2, which is a race condition that depends on the timing of whether the garbage collector deletes the object before the remediator has a chance to handle the previous update event from the applier, when the remediator was paused.

Reply:
Figured out the actual problem. The remediator wasn't reporting NoMatchErrors as conflicts. So it was only reporting a conflict when it got an (Object) NotFound error or Resource Version Conflict BEFORE the custom resource controller deleted the CR.